### DISCUSSION TITLE
Save multiband images of 2 bands in tiff format

### DISCUSSION THREAD

**Discussion by DavidStorm** [libvips/libvips#2259] [Q&A]
Hi,

I detect a problem saving multiband images of 2 bands in tiff format, photometric was changed from PHOTOMETRIC_SEPARATED to PHOTOMETRIC_MINISWHITE or PHOTOMETRIC_MINISBLACK in vips2tiff.c.

Line 755:
...
		if( wtiff-&gt;ready-&gt;Type == VIPS_INTERPRETATION_B_W ||
			wtiff-&gt;ready-&gt;Type == VIPS_INTERPRETATION_GREY16 ||
			wtiff-&gt;ready-&gt;Bands &lt; 3 ) { 
...

To keep the correct photometric, just need to verify the original image type as:
...
		if( wtiff->ready->Type == VIPS_INTERPRETATION_B_W ||
			wtiff->ready->Type == VIPS_INTERPRETATION_GREY16 ||
			(wtiff->ready->Bands < 3 && wtiff->ready->Type != VIPS_INTERPRETATION_MULTIBAND)) {
...

**Comment by jcupitt** [comment]
Hi @DavidStorm, thanks for the patch suggestion.

Could you share a sample image that shows this problem? I'm not sure I fully understand what's going on.

**Reply by DavidStorm** [reply]
https://github.com/libvips/libvips/discussions/2259#discussioncomment-756870

**Comment by DavidStorm** [comment]
Hi @jcupitt

I'm sending two files. Original file is 19073000.tif (Multichannel with 2 channels),  just open and save to Temp file TEMP_19073000.tif, you will get different photometric information.

<img width="980" alt="Captura de pantalla 2021-05-19 a las 9 52 34" src="https://user-images.githubusercontent.com/15983218/118776822-ad732680-b888-11eb-949c-966fc4759799.png">
<img width="976" alt="Captura de pantalla 2021-05-19 a las 9 52 47" src="https://user-images.githubusercontent.com/15983218/118776831-afd58080-b888-11eb-8d49-7f9c10f55fdf.png">

[Archivo comprimido.zip](https://github.com/libvips/libvips/files/6506887/Archivo.comprimido.zip)

**Comment by jcupitt** [comment]
Ah I see. Yes, libvips does not really support plane separated images, it always converts to interleaved.

I suppose we could add a flag to the TIFF writer for "please save in plane-separate format", but performance would be really horrible.

libvips supports load of plane-separate TIFFs, but only as a nod to compatibility, and they are handled rather poorly. You'll find loading a plane-separate image is much, much slower. 

Could you explain why you need a plane-separated image? Perhaps there's some other solution.

**Reply by DavidStorm** [reply]
Loading TEMP_19073000.tif I get image type VIPS_INTERPRETATION_B_W but the original file 19073000.tif was VIPS_INTERPRETATION_MULTIBAND
